scipy/stats/tests/gen_studentized_range_mpmath_ref.py

```diff
@@ -0,0 +1,186 @@
+# To run this script, run
+# `python gen_studentized_range_mpmath_ref.py`
+# in the "scipy/stats/tests/" directory
+
+# This script generates a JSON file "./data/studentized_range_mpmath_ref.json"
+# that is used to compare the accuracy of `studentized_range` functions against
+# precise (20 DOP) results generated using `mpmath`.
+
+# Note: I would have prefered to use pickle rather than JSON, but -
+# due to security concerns - decided against it.
+
+from collections import namedtuple
+import json
+import time
+
+from mpmath import gamma, pi, sqrt, quad, inf, mpf, mp, exp
+from mpmath import npdf as phi
+from mpmath import ncdf as Phi
+
+results_filepath = "data/studentized_range_mpmath_ref.json"
+
+MPResult = namedtuple("MPResult", ["src_case", "mp_result"])
+
+CdfCase = namedtuple("CdfCase",
+                     ["q", "k", "v", "expected_atol", "expected_rtol"])
+
+MomentCase = namedtuple("MomentCase",
+                        ["m", "k", "v", "expected_atol", "expected_rtol"])
+
+general_atol = 1e-11
+general_rtol = 1e-11
+mp.dps = 24
+
+cdf_cases = [
+    CdfCase(0.1, 3, 10, general_atol, general_rtol),
+    CdfCase(1, 3, 10, general_atol, general_rtol),
+    CdfCase(10, 3, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 10, general_atol, general_rtol),
+    CdfCase(1, 10, 10, general_atol, general_rtol),
+    CdfCase(10, 10, 10, general_atol, general_rtol),
+
+    CdfCase(0.1, 3, 100, general_atol, general_rtol),
+    CdfCase(1, 3, 100, general_atol, general_rtol),
+    CdfCase(10, 3, 100, general_atol, general_rtol),
+
+    CdfCase(0.1, 10, 100, general_atol, general_rtol),
+    CdfCase(1, 10, 100, general_atol, general_rtol),
+    CdfCase(10, 10, 100, general_atol, general_rtol),
+
+    CdfCase(1, 3, 1000, general_atol, general_rtol),
+    CdfCase(10, 3, 1000, 1e-10, general_rtol),
```

Comment:
What happens in the cases that appear to be missing?
It would be nice to generate all these with `itertools.product`.

Reply:
Adding these. I manually hardcoded the tests in an array rather than using itertools to handle special cases (like `10, 3, 1000`) which need slightly looser tolerances.